projects/bms_carrier/test/test_current_sense.c

```diff
@@ -0,0 +1,113 @@
+#include <string.h>
+
+#include "ads1259_adc.h"
+#include "bms.h"
+#include "current_sense.h"
+#include "delay.h"
+#include "exported_enums.h"
+#include "interrupt.h"
+#include "soft_timer.h"
+#include "test_helpers.h"
+
+#define TEST_CS_CONV_DELAY 10
+
+static SpiSettings s_spi_settings = {
+  .baudrate = 6000000,
+  .mosi = { .port = GPIO_PORT_B, 15 },
+  .miso = { .port = GPIO_PORT_B, 14 },
+  .sclk = { .port = GPIO_PORT_B, 13 },
+  .cs = { .port = GPIO_PORT_B, 12 },
+};
+
+static CurrentReadings s_reads;
+static uint8_t s_fault_bps_bitmask = 0;
+static bool s_fault_bps_clear = false;
+
+static double s_ads_read = 0.0;
+static Ads1259ErrorHandlerCb s_ads_cb = NULL;
+
+StatusCode TEST_MOCK(fault_bps)(uint8_t fault_bitmask, bool clear) {
+  s_fault_bps_bitmask = fault_bitmask;
+  s_fault_bps_clear = clear;
+  return STATUS_CODE_OK;
+}
+
+StatusCode TEST_MOCK(ads1259_get_conversion_data)(Ads1259Storage *storage) {
+  storage->reading = s_ads_read;
+  return STATUS_CODE_OK;
+}
+
+StatusCode TEST_MOCK(ads1259_init)(Ads1259Storage *storage, Ads1259Settings *settings) {
+  s_ads_cb = settings->handler;
+  return STATUS_CODE_OK;
+}
+
+void setup_test(void) {
+  interrupt_init();
+  soft_timer_init();
+  memset(&s_reads, 0, sizeof(s_reads));
+  s_fault_bps_bitmask = 0;
+  s_fault_bps_clear = false;
+
+  s_ads_read = 0.0;
+  s_ads_cb = NULL;
+}
+
+void teardown_test(void) {}
+
+void print_reads() {
+  for (int i = 0; i < NUM_STORED_CURRENT_READINGS; i++) {
+    printf("%d: %d\n", i, s_reads.readings[i]);
+  }
+  printf("avg: %d\n", s_reads.average);
+}
+
+void test_is_charging(void) {
+  s_ads_read = 0.5;
+  TEST_ASSERT_OK(current_sense_init(&s_reads, &s_spi_settings, TEST_CS_CONV_DELAY));
```

Comment:
You may want a way to cancel the soft timer at the end of the tests since I'm pretty sure currently the timer will just keep running throughout all of the test cases.

Reply:
I'm fairly certain timers are zeroed out upon `soft_timer_init()` in `setup_test`, I don't think this is an issue